src/components/swap/SwapBox.vue

```diff
@@ -130,7 +135,30 @@ export default defineComponent({
     },
     async swapFrom(model) {
       if (model) {
-        // TODO: Call the method that gets the gas fee and calculate swapTo.value
+        this.hasAllowance = false;
+        let balance;
+        // native tokens needs to approval?
+        if (model.type === "NATIVE") {
+          balance = await this.web3.eth.getBalance(this.account);
+
+          // this.swapFrom.balance = this.web3.utils.fromWei(this.account); // todo: why this doesnt work?
+          this.swapFrom.balance = new BigNumber(balance).shiftedBy(
```

Comment:
don't make a hardcoded 18 decimals as default, just leave it to give an error.

Reply:
removed and add a check to throw error if no decimals present